### DISCUSSION TITLE
Change "validate"?

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#671] [Q&A]
I was reviewing some AaC related material with a System Engineer recently and got an interesting bit of feedback..."Validate means something different to System Engineers than what you're using it for."  And honestly, I can't disagree with that statement.  To a system engineer, validation tends to refer to ensuring the system you've defined actually provides the desired value to the end user within the context (or CONOPS if you prefer) as intended.  This is separate and distinct from verification which tends to mean the system you built actually passes the defined requirements.

At first I just thought..."meh, not a big deal", but it did create a seed of concern.  Something like this could be an argument against adoption of a technology like AaC.

So, should we consider changing our vocabulary away from the command "validate" and language term "validator"?  Below are some alternative terms to consider (and my initial opinion on each).

Potential Command Options:
- check (Aligns with the Ansible ```--check``` inspiration of IaC)
- test (May be confusing because we're really testing the model definition, not the defined system)
- qa (Pretty generic, hard to imagine anyone arguing about performing qa)

Potential Language Options:
- checker (Just feels wrong to me)
- test (I worry about creating confusion with acceptance tests)
- constraint (I kind of like this one as the validators we currently use actually define model constraints more than anything)

If I were to choose I'd go with "check" and "constraint".  I think these are the simplest, most direct expression of what we're doing and they shouldn't result in vocabulary arguments.

So what do you think?  Should we consider some name changes to avoid potential confusion or conflict in the future?  Are the suggestions reasonable?  Do you have other suggestions to consider?  If you could choose anything, what would you choose?

**Comment by Coffee2Bits** [comment]
My preferred alternative would probably be `check` and `constraint`. Primitive constraints and definition constraints are a bit clearer than primitive validator and definition validator. I've found the alliteration of saying, "validators validating..." a bit tiring and hard to follow in conversation, but `check` and `constraint` feel like they'd be easier to follow.

**Comment by lizzcondrey** [comment]
I agree that "constraint" best matches what a current validator's primary goal is for being used on a model. But as far as command, I think leaning into the distinction you made initially of validation versus verification. Currently our validators are doing more of a verification of the defined system meeting the requirements (constraints of operation) of the system.

So, I propose command of "verify" and language of "constraint".
